ajo/personalization-server-side/README.md

```diff
@@ -0,0 +1,210 @@
+
+```javascript
+fetch(
+  "https://edge.adobedc.net/ee/v2/interact?dataStreamId=abc&requestId=123",
+  {
+    headers: {
+      accept: "*/*",
+      "accept-language": "en-US,en;q=0.9",
+      "cache-control": "no-cache",
+      "content-type": "text/plain; charset=UTF-8",
+      pragma: "no-cache",
+      "sec-fetch-dest": "empty",
+      "sec-fetch-mode": "cors",
+      "sec-fetch-site": "cross-site",
+      "sec-gpc": "1",
+      "Referrer-Policy": "strict-origin-when-cross-origin",
+      Referer: "https://localhost/",
+    },
+    body: JSON.stringify({
+      event: {
+        xdm: {
+          eventType: "decisioning.propositionFetch",
+          web: {
+            webPageDetails: {
+              URL: "https://localhost/",
+            },
+            webReferrer: {
+              URL: "",
+            },
+          },
+          identityMap: {
+            FPID: [
+              {
+                id: "xyz",
+                authenticatedState: "ambiguous",
+                primary: true,
+              },
+            ],
+          },
+          timestamp: "2022-06-23T22:21:00.878Z",
+        },
+        data: {},
+      },
+      query: {
+        identity: {
+          fetch: ["ECID"],
+        },
+        personalization: {
+          schemas: [
+            "https://ns.adobe.com/personalization/default-content-item",
+            "https://ns.adobe.com/personalization/html-content-item",
+            "https://ns.adobe.com/personalization/json-content-item",
+            "https://ns.adobe.com/personalization/redirect-item",
+            "https://ns.adobe.com/personalization/dom-action",
+          ],
+          surfaces: ["web://127.0.0.1/#sample-json-offer"]
+        },
+      },
+      meta: {
+        state: {
+          domain: "localhost",
+          cookiesEnabled: true,
+          entries: [
+            {
+              key: "kndctr_XXX_AdobeOrg_identity",
+              value: "abc123",
+            },
+            {
+              key: "kndctr_XXX_AdobeOrg_cluster",
+              value: "or2",
+            },
+          ],
+        },
+      },
+    }),
+    method: "POST",
+  }
+).then((res) => res.json());
+```
+
+4. The target offer from the form-based activity is read from the response and used when producing the HTML response.
+5. For form-based activities, display events must manually be sent in the implementation to indicate when the offer has been displayed. In this example the notification is sent server-side during the request lifecycle.
```

Comment:
"For code-based experience actions, ...."

Reply:
Done